packages/ez-react/src/components/StackedBars.tsx

```diff
@@ -68,19 +51,45 @@ export const StackedBars: FC<StackedBarsProps> = ({
   ]);
 
   return (
-    <g {...rest}>
-      {yDomainKeys.map((yDomainKey) => {
-        // @ts-ignore
-        const shapeData = scaledData[yDomainKey] as RectangleDatum[];
-        const color = colorScale.scale(yDomainKey);
-        return shapeData.map((shapeDatum: RectangleDatum, index: number) => {
-          return (
-            <Bar
-              key={`${yDomainKey}${index}`}
-              shapeDatum={{ ...shapeDatum, color }}
-            />
-          );
-        });
+    <g className="ez-stacked-bars" {...rest}>
+      {data.map((_datum, idx) => {
+        return (
+          // The Domain keys still needs to be sorted.
+          // We create a bar for every data row
+          // Each bar is a stack bar where every element is a domain key.
+          <g className="ez-stacked-bar">
+            {multiDomainKeys.map((yDomainKey, domainIdx) => {
+              const color = colorScale.scale(yDomainKey);
+              const shapeDatum =
+                // The height of the current bar will be computed
+                // the height will be currentDKHeight - previousDKHeight
+                // The first domain key will not be affected.
+                domainIdx !== 0
+                  ? {
+                      ...scaledDataDict[yDomainKey][idx],
+                      width:
+                        direction === Direction.HORIZONTAL
+                          ? scaledDataDict[yDomainKey][idx].width -
+                            scaledDataDict[multiDomainKeys[domainIdx - 1]][idx]
```

Comment:
```suggestion
                            scaledDataDict[multiDomainKeys[domainIdx - 1]][idx]
```
Let refactor this to avoid repetition.